contributors/design-proposals/storage/volume-topology-scheduling.md

```diff
@@ -276,6 +278,109 @@ StorageClasses for the volume types that need the new binding behavior.
 behavior depending on the StorageClass configuration.  We will mitigate this by
 adding a new PVC event to indicate if binding will follow the new behavior.
 
+#### Add a way to override topology in StorageClass
+Provisioning related topology information is currently specified as elements
+of parameters in StorageClass, which is obscure to users. Here we extend
+StorageClass with a new NodeAffinity field that specifies
+the topology constraints, like we does for PersistentVolume:
+
+```
+type StorageClass struct {
+    ...
+    NodeAffinity *VolumeNodeAffinity
+}
+```
+
+Thus, in predicates, the scheduler would check with the topology affinity
+information along with existing NodeAffinity information from the pod when making scheduling decision.
+
+_TBD: the common NodeSelector could hardly cover use case of `replica-zones` and `replica-zones:auto`,
+which are largely storage-specific. To support the cases, we may need to have a
+modification on the field, or even create a whole new struct for this.
+I've not think it clearly, @msau42 would like to hear your opinion if you already have a mature idea.
+Besides, currently I seprerate the capacity topologyKey from the
+NodeAffinity field, for I think the scope of the two could vary. Anyway,
+If we decide to manage them in a single field, again we need to have a modification
+on the field, because currently a selector may contain a list of selectorTerms,
+it’s hard to get a single key from it._
+
+
+#### Store topology-key related capacity in StorageClass
+For local storage and other potential volume types whose capacity is divided
```

Comment:
I haven't had a chance to look into the rest of this yet.  Will do so tomorrow.

I just pushed an update for the admin restriction use case.

Reply:
Actually, I took a look at the capacity reporting now, and pushed another update.  I did some rewording and also moved topology key to inside the capacity.  Let me know if that's going to have problems for the implementation.  I'm going to look at design details tomorrow.

Reply:
As we discussed, the topology key filed is thought to be a sigh on whether a storage's capacity is topology-aware. We should make it immutable after API object creation.
So in my opinion, the field is more of a first class field than that of status.

Reply:
Where is it going to be used though?  Is it only going to be used in the scheduler predicate to add up how much we've allocated out of the reported capacity?

Reply:
Agree we should be especially careful to add first class field or field in spec,
I just feel it weird to restrict modification on fields in status (if you agree to make it immutable)

Reply:
Yeah, making a status field immutable is strange, but I think we should think about the details a little more in depth before deciding where the right place is.  Right now, I imagine that the provisioner is going to be the one setting the topology key, not the admin.

Reply:
Yep, for most storage that has a central provisioner, it's ok to leave it to the provisioner,
but for local storage, whose provision plugins spread around the nodes, I think it better to let admin take control of it, also, proctection on the field might be necessary